sdk/src/Services/DynamoDBv2/Custom/DataModel/_async/Context.Async.cs

```diff
@@ -41,9 +41,12 @@ public partial class DynamoDBContext : IDynamoDBContext
         /// <param name="value">Object to save.</param>
         /// <param name="cancellationToken">Token which can be used to cancel the task.</param>
         /// <returns>A Task that can be used to poll or wait for results, or both.</returns>
-        public Task SaveAsync<T>(T value, CancellationToken cancellationToken = default(CancellationToken))
+        public async Task SaveAsync<T>(T value, CancellationToken cancellationToken = default(CancellationToken))
         {
-            return SaveHelperAsync(value, null, cancellationToken);
+            using (DynamoDBTelemetry.CreateSpan(this, nameof(SaveAsync)))
+            {
+                await SaveHelperAsync(value, null, cancellationToken).ConfigureAwait(false);
```

Comment:
Wouldn't it be better if you just create the span inside `SaveHelperAsync` instead of duplicating the code in the 4 `SaveAsync` overloads?

Reply:
This comment applies to all the other operations

Reply:
I though about this too but I am worried that in the future when we expand this library, we may add a new public composite operation that uses all these private methods and we'd end up setting the wrong operation name in the traces.

For example, if we create the trace inside `SaveHelperAsync` we'd still want to set the operation name to `SaveAsync` since that is the public API that customers use. 

In the future if we create a new composite operation called `Foo` that calls `LoadHelpersAsync`, manipulates the returned value and then again calls `SaveHelpersAsync`, we'd end up creating traces for `LoadAsync` and `SaveAsync` even though these operations were not called directly.